nbm-maven-plugin/pom.xml

```diff
@@ -135,8 +135,18 @@ under the License.
             <groupId>org.apache.maven.shared</groupId>
             <artifactId>maven-dependency-analyzer</artifactId>
             <version>1.13.2</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.ow2.asm</groupId>
+                    <artifactId>asm</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.ow2.asm</groupId>
+            <artifactId>asm</artifactId>
+            <version>9.7</version>
         </dependency>
```

Comment:
is the exclusion required? I thought a direct dependency would win over dependencies deeper down the tree.

Reply:
but you might be right, maven does sometimes downgrade versions

Reply:
With exclusion I would say I'm sure

Reply:
i was curious and checked since I always forget if the exclude is needed: version upgrade works without exclude